mathics/builtin/distance/numeric.py

```diff
@@ -0,0 +1,230 @@
+"""
+Numerial Data
+"""
+
+from mathics.builtin.base import Builtin
+from mathics.core.expression import Expression
+
+from mathics.core.symbols import (
+    SymbolAbs,
+    SymbolDivide,
+    SymbolPlus,
+    SymbolPower,
+    SymbolTimes,
+)
+from mathics.core.systemsymbols import (
+    SymbolMax,
+    SymbolNorm,
+    SymbolSubtract,
+    SymbolTotal,
+)
+
+
+def _norm_calc(head, u, v, evaluation):
+    expr = Expression(head, u, v)
+    old_quiet_all = evaluation.quiet_all
+    try:
+        evaluation.quiet_all = True
+        expr_eval = expr.evaluate(evaluation)
+    finally:
+        evaluation.quiet_all = old_quiet_all
+    if expr_eval.sameQ(expr):
+        evaluation.message("Norm", "nvm")
+        return None
+    else:
+        return expr_eval
+
+
+class BrayCurtisDistance(Builtin):
+    """
+    <dl>
+      <dt>'BrayCurtisDistance[$u$, $v$]'
+       <dd>returns the Bray-Curtis distance between $u$ and $v$.
+    </dl>
+
+    The Bray-Curtis distance is equivalent to Total[Abs[u-v]]/Total[Abs[u+v]].
+    >> BrayCurtisDistance[-7, 5]
+     = 6
+
+    >> BrayCurtisDistance[{-1, -1}, {10, 10}]
+     = 11 / 9
+    """
+
+    summary_text = "Bray-Curtis distance"
+
+    def apply(self, u, v, evaluation):
+        "BrayCurtisDistance[u_, v_]"
+        t = _norm_calc("Subtract", u, v, evaluation)
+        if t is not None:
+            return Expression(
+                SymbolDivide,
+                Expression("Total", Expression(SymbolAbs, t)),
+                Expression(
+                    "Total", Expression(SymbolAbs, Expression(SymbolPlus, u, v))
+                ),
+            )
+
+
+class CanberraDistance(Builtin):
+    """
+    <dl>
+      <dt>'CanberraDistance[$u$, $v$]'
+       <dd>returns the canberra distance between $u$ and $v$, which is a weighted version of the Manhattan distance.
+    </dl>
+
+    >> CanberraDistance[-7, 5]
+     = 1
+
+    >> CanberraDistance[{-1, -1}, {1, 1}]
+     = 2
+    """
+
+    summary_text = "Canberra distance"
+
+    def apply(self, u, v, evaluation):
+        "CanberraDistance[u_, v_]"
+        t = _norm_calc("Subtract", u, v, evaluation)
+        if t is not None:
+            return Expression(
+                "Total",
```

Comment:
SymbolTotal

Reply:
And other places in this fine. Done now.